Environment/Environment.tex

```diff
@@ -620,27 +628,53 @@ \subsection{Packages: declarative categorization of Squeak code}
+First, the method would be displayed as missing in the original package, so the package would unintentionally become dirty.
+Second, if you unloaded your own package again, the original method would be lost.
+
+A better way to do this is to mark the method as an \ind{overriding method} by placing it in a protocol named \prot{*SqueakLink-menu-override}.
+In general, an override protocol is an extension protocol that ends with the suffix \prot{-override}.
+Usually, it's a good idea to repeat the name of the original protocol in the override protocol name.
+
+However, beware that every change to an existing package might break other packages that depend on it.
+Thus, use override methods as seldom as possible.
+Often, there are better ways to extend the functionality of a package, such as subclassing or using explicit hooks.\footnote{Actually, there is a better way to extend the menu in our example: Place a new (non-override) extension method named \ct{squeakLinkMenu:} in the \ct{Inspector} class and add the pragma \ct{<fieldListMenu>} to it.}
+
+\subsubsection{Surveying the contents of a package}
 
-The way that you do this is by placing those methods in a protocol (of \ct{Object}, \ct{String}, \ct{Date}, and so on) named \prot{*SqueakLink} (note the initial asterisk).
-The combination of the \scat{SqueakLink-...} categories and the \prot{*SqueakLink} protocols form a package named \ct{SqueakLink}.
+The combination of the \scat{SqueakLink-...} categories and the \prot{*SqueakLink-...} protocols form a package named \ct{SqueakLink}.
 To be precise, the rules for what goes in a package are as follows.
 
 A package named \ct{Foo} contains:
 
-\begin{enumerate}		\label{sec:packageRules}
-	\item{} all class definitions of classes in the category \scat{Foo}, or in categories with names starting with \scat{Foo-},
+\begin{enumerate}\label{sec:packageRules}
+	\item{} all class definitions of classes in the category \scat{Foo}, or in categories with names starting with \scat{Foo-} (or \scat{foo-}, \scat{fOO}, autc., as Monticello actually always ignores the case of the letters),
 	and
-	\item{} \label{env:extensions} all method definitions in any class in a protocol named \prot{*Foo} or whose name starts with \prot{*Foo-} (when performing this name comparison, the case of the letters in the names is actually ignored),
+	\item{} \label{env:extensions} all method definitions in any class in a protocol named \prot{*Foo} or whose name starts with \prot{*Foo-},
 	and
 	\item{} all methods in classes in the category \scat{Foo} or in a category whose name starts with \scat{Foo-}, \emph{except} for those methods in protocols whose names start with \prot{*}.
 \end{enumerate}
 \noindent
 A consequence of these rules is that each class definition and each method belongs to exactly one package.
+% ct: This is NOT true if you define multiple packages that overlap each other, e.g., Kernel and Kernel-Exceptions! Should we mention that?

```

Comment:
That's a tough one. It would require rewritng this whole section on the basis that packages are an implicit concept that solely exists through the naming scheme. 
In practice, for beginners this does not make a large difference. Also, you could argue that Kernel is a package including the Kernel-* packages, so the method or class does actually belong to exactly one "most-specific" package, and potentially multiple package-packages. I vote for not discussing this.

Reply:
```suggestion
% ct: In fact, e.g., Kernel and Kernel-Exceptions and could be defined as overlapping packages. However, this is commonly not an intended usage of packages so we omit it for sake of simplicity here.
```